micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/CompilationMetrics.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jvm;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+
+import java.lang.management.CompilationMXBean;
+import java.lang.management.ManagementFactory;
+
+import static java.util.Collections.emptyList;
+
+@NonNullApi
+@NonNullFields
+public class CompilationMetrics implements MeterBinder {
```

Comment:
@shakuzen @jkschneider Should this class be named `JvmCompilationMetrics` since these are metrics around the JVM?

Reply:
I don't mind renaming the class but there is already a  `ClassLoaderMetrics` which should be renamed `JvmClassLoaderMetrics` in that case.

Reply:
I agree `JvmCompilationMetrics` is more clear here. `ClassLoaderMetrics` is already released public API so we can't change that without breaking all users of it.